src/ota_metadata/legacy/parser.py

```diff
@@ -659,7 +660,8 @@ def _process_metadata_jwt(self) -> _MetadataJWTClaimsLayout:
         with NamedTemporaryFile(prefix="metadata_jwt", dir=self.run_dir) as meta_f:
             _downloaded_meta_f = Path(meta_f.name)
 
-            while not _shutdown:
+            _retry_counter = 0
+            while _retry_counter < cfg.DOWNLOAD_GROUP_INACTIVE_TIMEOUT:
```

Comment:
this will be `_retry_counter * self.retry_interval < cfg.DOWNLOAD_GROUP_INACTIVE_TIMEOUT`